src/main/scala/tech/sourced/featurext/Client.scala

```diff
@@ -9,31 +9,70 @@ import tech.sourced.featurext.generated.service._
 import org.slf4j.{Logger => Slf4jLogger}
 import tech.sourced.gemini.util.MapAccumulator
 
-import scala.concurrent.Await
+import scala.concurrent.{Await, Future}
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.duration.{Duration, SECONDS}
 import scala.util.control.NonFatal
 
+
+sealed abstract class Extractor {
+  val Threshold = 5
+  def extract(client: FeatureExtractor, uast: Node): Future[FeaturesReply]
+  def extractFeatures(client: FeatureExtractor, uast: Node): Future[Seq[Feature]] = {
+    extract(client, uast).map(_.features) //avoids duplication of FeaturesReply->Feature code
+  }
+}
+
+case class IdentifiersExt(weight: Int, split: Boolean) extends Extractor {
+  override def extract(client: FeatureExtractor, uast: Node): Future[FeaturesReply] = {
+    val req = IdentifiersRequest(uast = Some(uast), docfreqThreshold = Threshold, weight = weight, splitStem = split)
+    client.identifiers(req)
+  }
+}
+
+case class GraphletExt(weight: Int) extends Extractor {
+  override def extract(client: FeatureExtractor, uast: Node): Future[FeaturesReply] = {
+    val req = GraphletRequest(uast = Some(uast), docfreqThreshold = Threshold, weight = weight)
+    client.graphlet(req)
+  }
+}
+
+case class LiteralsExt(weight: Int) extends Extractor {
+  override def extract(client: FeatureExtractor, uast: Node): Future[FeaturesReply] = {
+    val req = LiteralsRequest(uast = Some(uast), docfreqThreshold = Threshold, weight = weight)
+    client.literals(req)
+  }
+}
+
+case class Uast2seqExt(weight: Int, seqLen: Int, stride: Int) extends Extractor {
+  override def extract(client: FeatureExtractor, uast: Node): Future[FeaturesReply] = {
+    val req = Uast2seqRequest(
+      uast = Some(uast), docfreqThreshold = Threshold, weight = weight
+      //TODO(bzz): add seqLen = seqLen, stride = stride
+    )
+    client.uast2Seq(req)
+  }
+}
+
 object FEClient {
-  val idWeight = 194
-  val literalsWeight = 264
-  val graphletWeight = 548
 
-  def extract(uast: Node, client: FeatureExtractor, log: Slf4jLogger, skippedFiles: Option[MapAccumulator] = None):
-  Iterable[Feature] = {
-    val idRequest = IdentifiersRequest(uast=Some(uast), docfreqThreshold=5, weight = idWeight, splitStem = true)
-    val litRequest = LiteralsRequest(uast=Some(uast), docfreqThreshold=5, weight = literalsWeight)
-    val graphletRequest = GraphletRequest(uast=Some(uast), docfreqThreshold=5, weight = graphletWeight)
-    client.identifiers(idRequest)
+  val fileLevelExtractors = Seq(IdentifiersExt(194, split = true), GraphletExt(548), LiteralsExt(264))
```

Comment:
Even if it's clear what the arguments are, I'd personally prefer to see the vals defined as before, now it looks more like a magic number.

Reply:
Updated in d7185b1